sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorState.java

```diff
@@ -0,0 +1,172 @@
+public class ChangeFeedProcessorState {
+    private String id;
+    private String leaseToken;
+    private String hostName;
+    private Instant lastUpdate;
+    private String continuationToken;
+    private Instant continuationTokenTimestamp;
+    private int estimatedLag;
+
+    /**
+     * Gets the ID of the lease item representing the persistent state of a change feed processor worker.
+     *
+     * @return the ID of the lease item.
+     */
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Sets the ID of the lease item representing the persistent state of a chenge feed processor worker.
+     *
+     * @param id a unique string.
+     * @return the current ChangeFeedProcessorState instance.
+     */
+    public ChangeFeedProcessorState setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Gets the token of the lease item representing the persistent state of a change feed processor worker.
+     * <p>
+     * A lease token is a unique identifier representing a specific scope that a CFP worker will operate on.
+     *
+     * @return a string token uniquely representing the scope for one worker unit.
+     */
+    public String getLeaseToken() {
+        return this.leaseToken;
+    }
+
+    /**
+     * Sets the token of the lease item representing the persistent state of a change feed processor worker.
+     * <p>
+     * A lease token is a unique identifier representing a specific scope that a CFP worker will operate on.
+     *
+     * @param leaseToken a unique string representing a specific scope that a CFP worker will operate on.
+     * @return the current ChangeFeedProcessorState instance.
+     */
+    public ChangeFeedProcessorState setLeaseToken(String leaseToken) {
+        this.leaseToken = leaseToken;
+        return this;
+    }
+
+    /**
+     * Gets the name of the host which operates on the lease item.
+     * <p>
+     * When using multiple CFP instances distributing the work for a given feed container, each host must have a unique name.
+     *
+     * @return the host name that has ownership of this lease item or null if no host is currently operating on this lease.
+     */
+    public String getHostName() {
+        return this.hostName;
+    }
+
+    /**
+     * Sets the name of the host which operates on the lease item.
+     * <p>
+     * When using multiple CFP instances distributing the work for a given feed container, each host must have a unique name.
+     *
+     * @param hostName the host name that has ownership of this lease item.
+     * @return the current ChangeFeedProcessorState instance.
+     */
+    public ChangeFeedProcessorState setHostName(String hostName) {
+        this.hostName = hostName;
+        return this;
+    }
+
+    /**
+     * Gets the time when the lease item was last updated.
+     *
+     * @return time when the lease item was last updated.
+     */
+    public Instant getLastUpdate() {
```

Comment:
getLastUpdate vs getLastUpdateTime ?

Reply:
isn't that implicit given the return type?

Reply:
well, IMO the api method name should speak for itself without looking at the return type.

Reply:
np, I can make that change